mods/creative/init.lua

```diff
@@ -35,19 +40,29 @@ minetest.after(0, function()
 			end
 		end,
 	})
+
+	creative_inventory.update(player_name, nil)
+	--print("creative inventory size: "..dump(creative_inventory.creative_inventory_size))
+end
+
+function creative_inventory.update(player_name, filter)
 	local creative_list = {}
-	for name,def in pairs(minetest.registered_items) do
-		if (not def.groups.not_in_creative_inventory or def.groups.not_in_creative_inventory == 0)
-				and def.description and def.description ~= "" then
-			creative_list[#creative_list+1] = name
+	local inv = minetest.get_inventory({type = "detached", name = "creative_" .. player_name})
+
+	for name, def in pairs(minetest.registered_items) do
+		if not (def.groups.not_in_creative_inventory == 1) and
+				def.description and def.description ~= "" then
+			if (filter and def.name:find(filter)) or not filter then
```

Comment:
This `find` call searches for regular expressions instead of plain substrings. If this is intended, you have to wrap the `find` call in `pcall` and fail gracefully in case of errors. If not intended, change to `find(filter, 1, true)`.